### PR TITLE
[spi_device] Perf and Byte Access added

### PR DESCRIPTION
**Pull request by kosta-kojdic**
Added Perf and Byte access scenarios. Content is the same as #9376 which is closed due to commits conflict.

Signed-off-by: kosta-kojdic <kosta.kojdic@ensilica.com>

### CODE REVIEW

File: hw/ip/spi_device/dv/env/spi_device_env.core [hunk 1]
```diff
@@ -41,4 +43,4 @@ targets:
     filesets:
       - files_dv
     generate:
-      - ral
+      - ral
```
Comment (weicaiyang) [inline hw/ip/spi_device/dv/env/spi_device_env.core:46]: nit: please change this to a linux/unix newline (\n)

File: hw/ip/spi_device/dv/spi_device_sim_cfg.hjson [hunk 2]
```diff
@@ -90,4 +100,4 @@
       tests: ["spi_device_smoke"]
     }
   ]
-}
+}
```
Comment (weicaiyang) [inline hw/ip/spi_device/dv/spi_device_sim_cfg.hjson:103]: ditto

**Review by weicaiyang — APPROVED** [review]
LGTM with a nit

**Comment by kosta-kojdic** [comment]
Updated with lines at the end of files

**Pull request merged**